packages/paste-core/components/combobox/__tests__/MultiselectCombobox.spec.tsx

```diff
@@ -0,0 +1,235 @@
+  {group: 'Primitives', label: 'Box'},
+  {group: 'Primitives', label: 'Text'},
+  {group: 'Primitives', label: 'Non-modal dialog'},
+  {group: 'Layout', label: 'Grid'},
+  {label: 'Design Tokens'},
+];
+
+function getFilteredGroupedItems(inputValue: string): GroupedItem[] {
+  const lowerCasedInputValue = inputValue.toLowerCase();
+  return filter(groupedItems, (item: GroupedItem) => item.label.toLowerCase().includes(lowerCasedInputValue));
+}
+
+const GroupedMultiselectComboboxMock: React.FC<Partial<MultiselectComboboxProps>> = (props) => {
+  const [inputValue, setInputValue] = React.useState('');
+  const filteredItems = React.useMemo(() => getFilteredGroupedItems(inputValue), [inputValue]);
+
+  return (
+    <MultiselectCombobox
+      groupItemsBy="group"
+      items={filteredItems}
+      itemToString={(item: GroupedItem) => (item ? item.label : '')}
+      onInputValueChange={({inputValue: newInputValue = ''}) => {
+        setInputValue(newInputValue);
+      }}
+      labelText="Choose a component:"
+      helpText="This is group"
+      initialIsOpen
+      optionTemplate={(item: GroupedItem) => <div>{item.label}</div>}
+      {...props}
+    />
+  );
+};
+
+describe('MultiselectCombobox', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Render', () => {
+    it('should render a combobox with aria attributes', () => {
+      const {getByRole, getAllByRole} = render(<MultiselectComboboxMock initialIsOpen={false} />, {
+        wrapper: ThemeWrapper,
+      });
+
+      // PillGroup aria attributes
+      const [pillGroupListbox, dropdownListbox] = getAllByRole('listbox');
+      expect(pillGroupListbox.getAttribute('aria-label')).toBeDefined();
+      expect(pillGroupListbox.getAttribute('aria-describedby')).toBeDefined();
+
+      // Combobox aria attributes
+      const renderedCombobox = getByRole('combobox');
+      expect(renderedCombobox.getAttribute('aria-haspopup')).toEqual('listbox');
+      expect(renderedCombobox.getAttribute('aria-owns')).toEqual(dropdownListbox.id);
+      expect(renderedCombobox.getAttribute('aria-expanded')).toEqual('false');
+      fireEvent.click(renderedCombobox);
+      expect(renderedCombobox.getAttribute('aria-expanded')).toEqual('true');
+
+      // Textbox aria attributes
+      const renderedTextbox = getByRole('textbox');
+      expect(renderedTextbox.getAttribute('aria-controls')).toEqual(dropdownListbox.id);
+      expect(renderedTextbox.getAttribute('aria-describedby')).not.toEqual('');
+      expect(renderedTextbox.getAttribute('aria-labelledby')).toEqual(document.querySelector('label')!.id);
+
+      // unique option ids
+      const renderedOptions = getAllByRole('option');
+      const optionIDs = renderedOptions.map((option) => option.id);
+      const uniqueIDs = uniq(optionIDs);
+      expect(uniqueIDs.length).toEqual(optionIDs.length);
+    });
+
+    it('should render a label for that matches the input id', () => {
```

Comment:
Question: should this be combined with the "should render a combobox with aria attributes" test?